### PR TITLE
[ALLUXIO-2815] Remove thrift io exception

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2815

This PR removes ThriftIOException. We will always use AlluxioTException for passing exceptions through Thrift. We replace its usages with UnavailableException, which represents the same concept of retryable error.

### CODE REVIEW

File: core/common/src/main/java/alluxio/exception/status/AbortedException.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/exception/status/AbortedException.java]: I would expect these to be already merged in the exceptions branch?

File: core/common/src/thrift/exception.thrift [hunk 2]
```diff
@@ -1,12 +1,122 @@
...
+  UNAVAILABLE = 14,
+
+  // DataLoss indicates unrecoverable data loss or corruption.
+  DATA_LOSS = 15,
+}
+
 exception AlluxioTException {
   1: string type // deprecated since 1.1 and will be removed in 2.0
   2: string message
   3: string className
```
Comment (jsimsa) [inline core/common/src/thrift/exception.thrift]: should this have the status?

Comment (aaudiber) [inline core/common/src/thrift/exception.thrift]: It's added in https://github.com/Alluxio/alluxio/pull/5178

File: core/common/src/main/java/alluxio/exception/status/AbortedException.java (hunk 1 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/exception/status/AbortedException.java]: still waiting for review

Comment (jsimsa) [inline core/common/src/main/java/alluxio/exception/status/AbortedException.java]: I just merged it

**Comment by calvinjia** [comment]
@aaudiber What happens when a new client gets a `ThriftIOException` from an old server, and when an old client gets an `UnavailableException` from a new server?

**Comment by aaudiber** [comment]
@calvinjia I just bumped the API version so that old clients talking to new servers will see `FileSystemMasterClient client version 1 is not compatible with server version 2` and new clients talking to old servers will see `FileSystemMasterClient client version 2 is not compatible with server version 1`. I've confirmed this locally

**Pull request merged**